### PR TITLE
Improved GitVersionTask documentation

### PR DESCRIPTION
**Pull request by asbjornu**
Added more information to how GitVersionTask works, a little TL;DR section and rewrote a few section to make the documentation clearer. Addresses #789.

### CODE REVIEW

File: docs/usage/msbuild-task.md [hunk 1]
```diff
@@ -1,24 +1,49 @@
...
+The sub-task named `GitVersionTask.UpdateAssemblyInfo` will inject version metadata into the assembly
+which GitVersionTask is added to. For each assembly you want GitVersion to version, you need to
+install the GitVersionTask into it via NuGet.
 
-At build time a temporary AssemblyInfo.cs will be created that contains the appropriate SemVer information. This will will be included in the build pipeline.
+#### AssemblyInfo Attributes
 
-Ensure you remove the `Assembly*Version` attributes from your `Properties\AssemblyInfo.cs` file so as to not get duplicate attribute warnings. Sample default:
+At build time a temporary `AssemblyInfo.cs` will be created that contains the appropriate SemVer information.
+This will will be included in the build pipeline. Sample default:
```
Comment (gep13) [inline docs/usage/msbuild-task.md]: @asbjornu you have duplicated `will` here.

File: docs/usage/msbuild-task.md [hunk 2]
```diff
@@ -59,49 +84,59 @@ foreach (var field in fields)
...
-## Populate some MSBuild properties with version metadata
+### Populate some MSBuild properties with version metadata
 
-Task Name: `GitVersionTask.GetVersion`
+The sub-task `GitVersionTask.GetVersion` will write all the derived [variables](../more-info/variables.md)
+to MSBuild properties so the information can be used by other tooling in the build pipeline.
 
-At build time all the derived [variables](../more-info/variables.md) will be written to MSBuild properties so the information can be used by other tooling in the build pipeline.
+The class for `GitVersionTask.GetVersion` has a property for each variable. However at MSBuild time these
+properties a mapped to MSBuild properties that are prefixed with `GitVersion_`. This prevents conflicts with
```
Comment (gep13) [inline docs/usage/msbuild-task.md]: @asbjornu I realise this wasn't you, but can you change `these properties a mapped` to `these properties are mapped`

File: docs/usage/msbuild-task.md [hunk 3]
```diff
@@ -59,49 +84,59 @@ foreach (var field in fields)
...
-    <Message Text="GitVersion_InformationalVersion: $(GitVersion_InformationalVersion)"/> 
+```xml
+<Message Text="GitVersion_InformationalVersion: $(GitVersion_InformationalVersion)"/> 
+```
 
-## Communicate variables to current Build Server
+### Communicate variables to current Build Server
 
-Task Name: `GitVersionTask.WriteVersionInfoToBuildLog`
+The sub-task `GitVersionTask.WriteVersionInfoToBuildLog` will attemp to write the version information
```
Comment (gep13) [inline docs/usage/msbuild-task.md]: @asbjornu slight typo.  Should be `attempt`

File: docs/usage/msbuild-task.md [hunk 4]
```diff
@@ -59,49 +84,59 @@ foreach (var field in fields)
...
+to the current build server.
```
Comment (gep13) [inline docs/usage/msbuild-task.md]: @asbjornu should this be `the current build server log`?

File: docs/usage/msbuild-task.md [hunk 5]
```diff
@@ -59,49 +84,59 @@ foreach (var field in fields)
...
 
-If, at build time, it is detected that the build is occurring inside a Build Server server then the [variables](../more-info/variables.md) will be written to the build log in a format that the current Build Server can consume. See [Build Server Support](../build-server-support/build-server-support.md).
+If, at build time, it is detected that the build is occurring inside a Build Server server then the
```
Comment (gep13) [inline docs/usage/msbuild-task.md]: @asbjornu again, not you, but can you remove the duplicate `server`

**Comment by gep13** [comment]
@asbjornu looks good, just a couple small comments.  If you can fix up, I can merge it in. Thanks!

**Comment by asbjornu** [comment]
@gep13 Thanks for the review. Everything should be fixed.

**Comment by gep13** [comment]
LGTM! Thanks!

**Pull request merged**